.github/workflows/schemacode_ci.yml

```diff
@@ -5,92 +5,81 @@ on:
     branches:
       - "master"
       - "autodata2"
-    paths:
-      - "tools/schemacode/**"
-      - "src/schema/**"
     tags:
       - "schema-*"
   pull_request:
     branches:
       - "*"
-    paths:
-      - "tools/schemacode/**"
-      - "src/schema/**"
 
-jobs:
+defaults:
+  run:
+    shell: bash
 
-  run_tests:
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: ["ubuntu-latest"]
+        python-version: ["3.10"]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: "Install build dependencies"
+        run: pip install --upgrade build twine
+      - name: "Install test dependencies on tag"
+        run: pip install --upgrade pytest pyyaml pandas tabulate
+        if: ${{ startsWith(github.ref, 'refs/tags/schema-') }}
+      - name: "Build archive on tag"
+        run: pytest tools/schemacode/bidsschematools -k make_archive
+        env:
+          BIDSSCHEMATOOLS_RELEASE: 1
+        if: ${{ startsWith(github.ref, 'refs/tags/schema-') }}
+      - name: "Build source distribution and wheel"
+        run: python -m build tools/schemacode
+      - name: "Check distribution metadata"
+        run: twine check tools/schemacode/dist/*
+      - uses: actions/upload-artifact@v3
+        with:
+          name: dist
+          path: tools/schemacode/dist/
 
+  test:
     runs-on: ${{ matrix.os }}
+    needs: [build]
     strategy:
-        fail-fast: false
-        matrix:
-            os: ["ubuntu-latest"]
-            python-version: ["3.7", "3.8", "3.9", "3.10"]
+      fail-fast: false
+      matrix:
+        os: ["ubuntu-latest"]
+        python-version: ["3.7", "3.8", "3.9", "3.10"]
     name: ${{ matrix.os }} with Python ${{ matrix.python-version }}
-    defaults:
-      run:
-        shell: bash
     steps:
       - uses: actions/checkout@v3
 
       - name: "Set up Python"
         uses: actions/setup-python@v4
         with:
-            python-version: ${{ matrix.python-version }}
+          python-version: ${{ matrix.python-version }}
 
       - name: "Display Python version"
-        shell: bash {0}
         run: python -c "import sys; print(sys.version)"
 
-      - name: "Set Release Variable based on tag"
-        shell: bash {0}
-        run: |
-          MY_TAG="$(git describe --tags)"
-          if [ "${MY_TAG:0:7}" = "schema-" ]; then
-            echo "BIDSSCHEMATOOLS_RELEASE=1" >> $GITHUB_ENV
-          fi
+      - name: "Fetch packages"
+        uses: actions/download-artifact@v3
+        with:
+          name: dist
+          path: dist/
 
-      - name: "Install the schemacode package"
-        shell: bash {0}
+      - name: "Install package"
         run: |
-          python -m pip install --progress-bar off --upgrade pip
-          python -m pip install ./tools/schemacode[all]
+          pip install $( ls dist/*.whl )[all]
```

Comment:
Here we just install the built wheel. For regular builds, we'll continue to use git to fetch examples. For tag builds, we'll get the bundled test data.

Reply:
So the wheel will include the testdata if it's a tag build?

Reply:
Yes. And we can check the logs to verify that we're seeing it use the test data instead of hitting git.